// @mintlify/models@0.0.182/dist/mintconfig/localization.js
/*
Currently supported languages, should be ISO 639:
English - en
Chinese - cn, ISO 639 = zh
Simplified Chinese - zh-Hans
Traditional Chinese - zh-Hant
Spanish - es
French - fr
Japanese - jp, ISO 639 = ja
Portuguese - pt
Brazilian Portugese - pt-BR
German - de
Korean - ko
Italian - it
Russian - ru
Indonesian - id
Arabic - ar
Turkish - tr

*/
export const locales = [
    'en',
    'cn',
    'zh',
    'zh-Hans',
    'zh-Hant',
    'es',
    'fr',
    'ja',
    'jp',
    'pt',
    'pt-BR',
    'de',
    'ko',
    'it',
    'ru',
    'id',
    'ar',
    'tr',
];
export const localeDisplayNames = {
    en: 'English',
    cn: 'Chinese',
    zh: 'Chinese',
    'zh-Hans': 'Simplified Chinese',
    'zh-Hant': 'Traditional Chinese',
    es: 'Spanish',
    fr: 'French',
    ja: 'Japanese',
    jp: 'Japanese',
    pt: 'Portuguese',
    'pt-BR': 'Brazilian Portuguese',
    de: 'German',
    ko: 'Korean',
    it: 'Italian',
    ru: 'Russian',
    id: 'Indonesian',
    ar: 'Arabic',
    tr: 'Turkish',
};
export const localeDisplayFlags = {
    en: '🇺🇸',
    cn: '🇨🇳',
    zh: '🇨🇳',
    'zh-Hans': '🇨🇳',
    'zh-Hant': '🇹🇼',
    es: '🇪🇸',
    fr: '🇫🇷',
    ja: '🇯🇵',
    jp: '🇯🇵',
    pt: '🇧🇷',
    'pt-BR': '🇧🇷',
    de: '🇩🇪',
    ko: '🇰🇷',
    it: '🇮🇹',
    ru: '🇷🇺',
    id: '🇮🇩',
    ar: '🇸🇦',
    tr: '🇹🇷',
};
export const getLocaleDisplayName = (locale) => {
    return localeDisplayNames[locale];
};
export const getLocaleDisplayFlag = (locale) => {
    return localeDisplayFlags[locale];
};
export const LocaleCodes = Object.keys(localeDisplayNames);
export const LocaleDisplayNames = Object.values(localeDisplayNames);
export const LocaleDisplayFlags = Object.values(localeDisplayFlags);
export const LlmSupportedLocaleCodes = LocaleCodes.filter((locale) => !['jp', 'ja', 'ko', 'cn', 'zh', 'zh-Hans', 'zh-Hant', 'ru', 'ar'].includes(locale));
export const LlmSupportedLocaleDisplayNames = LocaleDisplayNames.filter((locale) => !['Japanese', 'Chinese', 'Russian', 'Arabic', 'Korean'].includes(locale));
export const LlmSupportedLocaleDisplayFlags = LocaleDisplayFlags.filter((locale) => !['🇯🇵', '🇨🇳', '🇷🇺', '🇸🇦', '🇰🇷'].includes(locale));
